lib/src/package_info.dart

```diff
@@ -0,0 +1,197 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:path/path.dart' as p;
+import 'package:pub_semver/pub_semver.dart';
+
+import 'package_name.dart';
+import 'system_cache.dart';
+
+/// Holds the package information
+abstract class PackageInfo {
```

Comment:
I think you can use PackageRange to represent this information.
Alternatively make it private in add.dart